src/MatrixFields/field_name_dict.jl

```diff
@@ -181,11 +181,51 @@ function get_internal_entry(
         entry
     elseif name_pair[2] == @name() && broadcasted_has_field(T, name_pair[1])
         # multiplication case 2 or 4, second argument
-        Base.broadcasted(entry) do matrix_row
-            map(matrix_row) do matrix_row_entry
-                broadcasted_get_field(matrix_row_entry, name_pair[1])
-            end
-        end # Note: This assumes that the entry is in a FieldMatrixBroadcasted.
+        target_field_eltype = broadcasted_get_field_type(T, name_pair[1])
+        if target_field_eltype <: Number
+            T_band = eltype(entry)
+            singleton_datalayout =
+                DataLayouts.singleton(Fields.field_values(entry))
+            # BandMatrixRow with same lowest diagonal and bandwidth as `entry`, with a scalar eltype
+            scalar_band_type = BandMatrixRow{
+                T_band.parameters[1],
+                T_band.parameters[2],
+                eltype(parent(entry)),
+            }
```

Comment:
```suggestion
            scalar_band_type =
                band_matrix_row_type(outer_diagonals(T_band)..., eltype(parent(entry)))
```
I think we should generally avoid accessing internals like this, since public API methods are easier to read/debug than `<type>.parameters`. If you think the suggested version is too clunky, though, feel free to add a new method to `MatrixFields` that computes this in a single function call.